src/commands/link.rs

```diff
@@ -143,176 +143,116 @@ fn select_environment(
-        Regex::new(r#"(?i)^[0-9A-F]{8}-[0-9A-F]{4}-4[0-9A-F]{3}-[89AB][0-9A-F]{3}-[0-9A-F]{12}"#)
-            .unwrap();
-    let team = match (project.as_ref(), team.as_ref()) {
-        (Some(project), None) if uuid_regex.is_match(project) => {
-            // It's a project id, figure out team
-            if me
-                .projects
-                .edges
-                .iter()
-                .any(|pro| pro.node.id.to_lowercase() == project.to_lowercase())
-            {
-                fake_select("Select a team", "Personal");
-                Team::Personal
-            } else if let Some(team) = me.teams.edges.iter().find(|team| {
-                team.node
-                    .projects
-                    .edges
-                    .iter()
-                    .any(|proj| proj.node.id.to_lowercase() == project.to_lowercase())
-            }) {
-                fake_select("Select a team", &team.node.name);
-                Team::Team(&team.node)
-            } else {
-                return Err(RailwayError::ProjectNotFound.into());
-            }
-        }
+    workspace_name: Option<String>,
+    response: queries::user_projects::ResponseData,
+) -> Result<Workspace, anyhow::Error> {
+    let workspace = match (project, workspace_name) {
         (Some(project), None) => {
-            // this means project name without team
-            if me.teams.edges.is_empty() {
-                // no teams, so it's personal
-                // if there is a project that has the same name
-                if me
-                    .projects
-                    .edges
-                    .iter()
-                    .any(|p| p.node.name.to_lowercase() == project.to_lowercase())
-                {
-                    fake_select("Select a team", "Personal");
-                    Team::Personal
-                } else {
-                    return Err(RailwayError::ProjectNotFound.into());
-                }
+            // It's a project id or name, figure out workspace
+            if let Some(workspace) = response.external_workspaces.iter().find(|w| {
+                w.projects.iter().any(|pro| {
+                    pro.id.to_lowercase() == project.to_lowercase()
+                        || pro.name.to_lowercase() == project.to_lowercase()
+                })
+            }) {
+                fake_select("Select a workspace", &workspace.name);
+                Workspace::External(workspace.clone())
+            } else if let Some(w) = response.me.workspaces.iter().find(|w| {
+                w.team.as_ref().map_or(false, |t| {
+                    t.projects.edges.iter().any(|pro| {
+                        pro.node.id.to_lowercase() == project.to_lowercase()
+                            || pro.node.name.to_lowercase() == project.to_lowercase()
+                    })
+                })
+            }) {
+                fake_select("Select a workspace", &w.name);
+                Workspace::Member(w.clone())
```

Comment:
Can this be combined with an `extend` like in `init`? We might want to abstract fetching the workspaces so commands don't have to worry about external/non-external workspaces. I think a `workspace.rs` file in the controllers ([like we have for the project](https://github.com/railwayapp/cli/blob/master/src/controllers/project.rs)) would be useful